src/endpoints/blocks/blocks.gateway.ts

```diff
@@ -0,0 +1,48 @@
+import { WebSocketGateway, WebSocketServer, SubscribeMessage, OnGatewayDisconnect } from '@nestjs/websockets';
+import { Server, Socket } from 'socket.io';
+import { BlockService } from './block.service';
+import { BlockFilter } from './entities/block.filter';
+import { QueryPagination } from 'src/common/entities/query.pagination';
+
+@WebSocketGateway({ cors: { origin: '*' } })
+export class BlocksGateway implements OnGatewayDisconnect {
+    @WebSocketServer()
+    server!: Server;
+
+    constructor(private readonly blockService: BlockService) { }
+
+    @SubscribeMessage('subscribeBlocks')
+    async handleSubscription(client: Socket, payload: any) {
```

Comment:
no validation on incoming websocket payloads and any malformed data can crash the service